src/mats_l1_processing/read_parquet_functions.py

```diff
@@ -278,10 +278,16 @@ def read_ccd_data_in_interval(
     )
     if filter != None:
         for variable in filter.keys():
-            filterlist &= (
-                (ds.field(variable) >= filter[variable][0])
-                & (ds.field(variable) <= filter[variable][1])
-            )
+            if type(filter[variable]) == list and len(filter[variable]) == 2:
```

Comment:
This could be replaced with the more compact `if isinstance(filter[variable], list) and ...`. That won't work for the `in (...)` below, so maybe not a good idea, but I thought you should know about it as an option.

Also note, that rather than `for variable in filter.keys():` you can use `for variable, value in filter.items():` so that you don't have to write `filter[variable]` everywhere.